### PR TITLE
Skip over lines that don't set an envvars

### LINKED ISSUE #285: Skip over non-envvar lines
`.env` files may contain lines that don't set environment variables. For example, I use [autoenv](https://github.com/kennethreitz/autoenv) in order to automatically activate the current project's Python virtual environment. I also set per-project aliases in `.env`. 

`dotenv` raises an error when it encounters lines that don't set envvars:

```
/usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv/parser.rb:63:in `parse_line': Line "source activate osf" doesn't match format (Dotenv::FormatError)
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv/parser.rb:47:in `block in call'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv/parser.rb:46:in `each'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv/parser.rb:46:in `call'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv/parser.rb:36:in `call'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv/environment.rb:13:in `load'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv/environment.rb:9:in `initialize'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:15:in `new'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:15:in `block (2 levels) in load'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:59:in `ignoring_nonexistent_files'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:14:in `block in load'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:46:in `block in with'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:45:in `each'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:45:in `reduce'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:45:in `with'
        from /usr/local/lib/ruby/gems/2.4.0/gems/dotenv-2.2.0/lib/dotenv.rb:13:in `load'
        from /usr/local/lib/ruby/gems/2.4.0/gems/docker-sync-0.2.0/lib/docker-sync/sync_manager.rb:19:in `initialize'
        from /usr/local/lib/ruby/gems/2.4.0/gems/docker-sync-0.2.0/tasks/sync/sync.thor:30:in `new'
        from /usr/local/lib/ruby/gems/2.4.0/gems/docker-sync-0.2.0/tasks/sync/sync.thor:30:in `start'
        from /usr/local/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor/command.rb:27:in `run'
        from /usr/local/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/local/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor.rb:369:in `dispatch'
        from /usr/local/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor/base.rb:444:in `start'
        from /usr/local/lib/ruby/gems/2.4.0/gems/docker-sync-0.2.0/bin/docker-sync:14:in `<top (required)>'
        from /usr/local/bin/docker-sync:22:in `load'
        from /usr/local/bin/docker-sync:22:in `<main>'
```

Would the maintainer(s) be open to skipping these lines rather than raising an error?

### PR DESCRIPTION
**Pull request by sloria**
This makes dotenv play better with .env files that
do more than just set variables, e.g. defining aliases
and functions

closes #285

### CODE REVIEW

File: lib/dotenv/parser.rb [hunk 1]
```diff
@@ -59,8 +59,6 @@ def parse_line(line)
         if variable_not_set?(line)
           raise FormatError, "Line #{line.inspect} has an unset variable"
         end
-      elsif line !~ /\A\s*(?:#.*)?\z/ # not comment or blank line
-        raise FormatError, "Line #{line.inspect} doesn't match format"
```
Comment (bkeepers) [inline lib/dotenv/parser.rb]: Would it make sense to `warn` on this?

Comment (sloria) [inline lib/dotenv/parser.rb]: I don't have a strong feeling about it, but I would think not. We don't expect every line to follow this format, so we shouldn't raise a warning.

Comment (siegy22) [inline lib/dotenv/parser.rb]: Maybe just warn once when the file parsing is finished :grin:?

**Comment by sloria** [comment]
@bkeepers Are there any further changes that need to be made before this is merged? Would you like me to add warning(s), or is this ok as is?

**Comment by sloria** [comment]
I still think this implements the correct behavior. Please let me know if there are any further changes to be made.

**Comment by jonmagic** [comment]
Hi @sloria, I'm taking over maintenance of dotenv with @cadwallion and @JessRudder and your PR is on our radar. Were you still interested in seeing this shipped? If so I will re-open, review, and respond or merge. Thank you for your time :smile:

**Comment by sloria** [comment]
@jonmagic Thanks for the response. I opened this issue because I ran into it when using [docker-sync](https://github.com/EugenMayer/docker-sync). See https://github.com/EugenMayer/docker-sync/issues/254 . 

I was able to work around the issue by changing the file that autoenv reads, but that's a bit of a hack. I I still think this PR implements the correct behavior, so I think you should consider reviewing/merging it.

**Comment by jonmagic** [comment]
@sloria excellent, thank you for the response.

**Comment by jonmagic** [comment]
We'll be cutting a new release next week that will include the changes from this PR.

**Pull request merged**